operationsgateway_api/src/records/echo_interface.py

```diff
@@ -0,0 +1,110 @@
+from io import BytesIO
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+
+from operationsgateway_api.src.config import Config
+from operationsgateway_api.src.exceptions import EchoS3Error
+
+
+log = logging.getLogger()
+
+
+class EchoInterface:
+    """
+    A class containing the functionality needed to contact Echo S3 object storage. In
+    OperationsGateway API, this is used to store and retrieve full-size images.
+
+    In boto3, the same exception class is used for all exceptions, `ClientError`. This
+    object has a `response` attribute storing a dictionary containing information of the
+    error. The typical format of this dictionary can be found at:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html
+    """
+
+    def __init__(self) -> None:
+        log.debug("Creating S3 resource to connect to Echo")
+        try:
+            self.resource = boto3.resource(
+                "s3",
+                endpoint_url=Config.config.images.echo_url,
+                aws_access_key_id=Config.config.images.echo_access_key,
+                aws_secret_access_key=Config.config.images.echo_secret_key,
+            )
+
+            log.debug("Retrieving bucket '%s'", Config.config.images.image_bucket_name)
+            self.bucket = self.resource.Bucket(Config.config.images.image_bucket_name)
+        except ClientError as exc:
+            log.error(
+                "%s: %s",
+                exc.response["Error"]["Code"],
+                exc.response["Error"]["Message"],
+            )
+            raise EchoS3Error(
+                "Error '%s' caught when creating a boto3 resource and retrieving a"
```

Comment:
This should probably be a bit more user friendly and not mention the bucket name or boto3. Maybe something like "Error retrieving object storage bucket" then add the code and the message.

Reply:
Fixed in [a475270](https://github.com/ral-facilities/operationsgateway-api/pull/60/commits/a475270bf8d776d6c611b8546d872a39f5f985d6)